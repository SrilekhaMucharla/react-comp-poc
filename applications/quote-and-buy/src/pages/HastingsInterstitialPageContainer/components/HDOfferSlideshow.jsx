import React from 'react';
import PropTypes from 'prop-types';
import { Carousel } from 'react-bootstrap';

function HDOfferSlideshow(props) {
    const { items, interval } = props;
    return (
        <Carousel fade interval={interval} pause={false} controls={false} indicators={false}>
            {(items && items.length > 0) && items.map((subtitle, index) => (
                <Carousel.Item key={`item no ${index + 1}`}>
                    {subtitle}
                </Carousel.Item>
            ))}
        </Carousel>
    );
}

HDOfferSlideshow.propTypes = {
    items: PropTypes.arrayOf(PropTypes.element).isRequired,
    interval: PropTypes.number
};

HDOfferSlideshow.defaultProps = {
    interval: 1000
};

export default HDOfferSlideshow;
