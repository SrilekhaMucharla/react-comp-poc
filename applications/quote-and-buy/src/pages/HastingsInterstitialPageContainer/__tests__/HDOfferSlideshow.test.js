import { mount } from 'enzyme';
import React from 'react';
import HDOfferSlideshow from '../components/HDOfferSlideshow';

const slideshowItems = [
    (<div>Test01</div>),
    (<div>Test02</div>),
    (<div>Test03</div>),
];

describe('<HDOfferSlideshow />', () => {
    test('component renders', () => {
        const wrapper = mount(<HDOfferSlideshow items={slideshowItems} />);
        expect(wrapper).toMatchSnapshot();
    });
});
