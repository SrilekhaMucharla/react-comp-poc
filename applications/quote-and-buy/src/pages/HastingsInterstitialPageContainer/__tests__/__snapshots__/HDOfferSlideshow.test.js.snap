// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDOfferSlideshow /> component renders 1`] = `
<HDOfferSlideshow
  interval={1000}
  items={
    Array [
      <div>
        Test01
      </div>,
      <div>
        Test02
      </div>,
      <div>
        Test03
      </div>,
    ]
  }
>
  <Carousel
    controls={false}
    defaultActiveIndex={0}
    fade={true}
    indicators={false}
    interval={1000}
    keyboard={true}
    nextIcon={
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
    }
    nextLabel="Next"
    pause={false}
    prevIcon={
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
    }
    prevLabel="Previous"
    slide={true}
    touch={true}
    wrap={true}
  >
    <div
      className="carousel slide carousel-fade"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="carousel-inner"
      >
        <Transition
          addEndListener={[Function]}
          appear={false}
          enter={true}
          exit={true}
          in={true}
          key=".$item no 1"
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          unmountOnExit={false}
        >
          <CarouselItem
            className="active"
            key="item no 1"
          >
            <div
              className="active carousel-item"
            >
              <div>
                Test01
              </div>
            </div>
          </CarouselItem>
        </Transition>
        <Transition
          addEndListener={[Function]}
          appear={false}
          enter={true}
          exit={true}
          in={false}
          key=".$item no 2"
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          unmountOnExit={false}
        >
          <CarouselItem
            className=""
            key="item no 2"
          >
            <div
              className="carousel-item"
            >
              <div>
                Test02
              </div>
            </div>
          </CarouselItem>
        </Transition>
        <Transition
          addEndListener={[Function]}
          appear={false}
          enter={true}
          exit={true}
          in={false}
          key=".$item no 3"
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          unmountOnExit={false}
        >
          <CarouselItem
            className=""
            key="item no 3"
          >
            <div
              className="carousel-item"
            >
              <div>
                Test03
              </div>
            </div>
          </CarouselItem>
        </Transition>
      </div>
    </div>
  </Carousel>
</HDOfferSlideshow>
`;
